spock-core/src/main/java/org/spockframework/mock/runtime/MockController.java

```diff
@@ -32,22 +32,32 @@ public MockController() {
   }
 
   @Override
-  public synchronized Object handle(IMockInvocation invocation) {
+  public Object handle(IMockInvocation invocation) {
```

Comment:
Maybe add a release notes entry, that the issue is fixed.

Reply:
For simple entries, I usually do a pass before the release to create any missing ones. Unfortunately, release notes are always a prime candidate for merge conflicts.